pkg/console/subresource/configmap/configmap_test.go

```diff
@@ -0,0 +1,286 @@
+					ResourceVersion:            "",
+					Generation:                 0,
+					CreationTimestamp:          metav1.Time{},
+					DeletionTimestamp:          nil,
+					DeletionGracePeriodSeconds: nil,
+					Labels:                     map[string]string{"app": controller.OpenShiftConsoleName},
+					Annotations:                nil,
+					OwnerReferences:            nil,
+					Initializers:               nil,
+					Finalizers:                 nil,
+					ClusterName:                "",
+				},
+				Data:       map[string]string{"console-config.yaml": exampleYaml},
+				BinaryData: nil,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := DefaultConfigMap(tt.args.cr, tt.args.rt); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("DefaultConfigMap() = %v\n ----------- want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestStub(t *testing.T) {
+	tests := []struct {
+		name string
+		want *corev1.ConfigMap
+	}{
+		{
+			name: "Testing Stub function configmap",
+			want: &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:                       ConsoleConfigMapName,
+					GenerateName:               "",
+					Namespace:                  controller.OpenShiftConsoleName,
+					SelfLink:                   "",
+					UID:                        "",
+					ResourceVersion:            "",
+					Generation:                 0,
+					CreationTimestamp:          metav1.Time{},
+					DeletionTimestamp:          nil,
+					DeletionGracePeriodSeconds: nil,
+					Labels:                     map[string]string{"app": controller.OpenShiftConsoleName},
+					Annotations:                nil,
+					OwnerReferences:            nil,
+					Initializers:               nil,
+					Finalizers:                 nil,
+					ClusterName:                "",
+				},
+				BinaryData: nil,
+				Data:       nil,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := Stub(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("\nStub() = %v\n -------- want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+// This unit test relies on both NewYamlConfig and NewYamlConfigString
+// to ensure the serialized data is created from host name
+// TODO: remove - This unit test is probably not useful since it is just testing yaml methods slice and marshal with no logic
+func TestNewYamlConfig(t *testing.T) {
+	type args struct {
+		host string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "TestNewYamlConfig",
+			args: args{
+				host: host,
+			},
+			want: exampleYaml,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewYamlConfigString(tt.args.host); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewYamlConfig() = \n%v\n ----> want\n%v", got, tt.want)
+			}
+		})
+	}
+}
+
+// TODO: remove - This unit test is not useful since it is just testing yaml method mapslice
```

Comment:
I agree, we can probably drop these. I think its important to test the public methods, and probably private that do some kind of logic.  this will be implicitly caught by testing `Default` and `Stub` funcs.